packages/web-client/app/components/card-pay/safe-balances/index.hbs

```diff
@@ -0,0 +1,74 @@
+<section class='card-pay-safe-balances__container' data-test-safe-balances>
+  <header
+    aria-label='metadata'
+    class='card-pay-safe-balances__header'
+    style={{css-var
+      safe-balances-header-background-color=(or this.data.info.backgroundColor 'var(--boxel-dark)')
+      safe-balances-header-text-color=(or this.data.info.textColor 'var(--boxel-light)')
+    }}
+    data-test-safe-balances-header
+  >
+    <div class='card-pay-safe-balances__network-and-address'>
+      <span class='card-pay-safe-balances__network' data-test-safe-balances-network>
+        On {{network-display-info 'layer2' 'shortName'}}
+      </span>
+      <span class='card-pay-safe-balances__address'>
+        {{truncate-middle @safe.address}}
+      </span>
+    </div>
+    <div class='card-pay-safe-balances__type' data-test-safe-balances-type>{{this.safeType}}</div>
+  </header>
+  <section class='card-pay-safe-balances__body'>
+    <header aria-label='title' class='card-pay-safe-balances__title-container' data-test-safe-balances-title>
+      {{#if this.data.icon}}
+        {{svg-jar this.data.icon width='20' height='20' class='card-pay-safe-balances__logo' data-test-safe-balances-logo=this.data.icon}}
+      {{else}}
+        <CardPay::MerchantLogo
+          class='card-pay-safe-balances__logo'
+          @name={{first-char this.data.info.name}}
+          @logoBackground={{this.data.info.backgroundColor}}
+          @logoTextColor={{this.data.info.textColor}}
+        />
```

Comment:
Merchant logo sometimes get squished here if the merchant's name is long. We should add some defensive CSS to the merchant logo:

```
  flex-shrink: 0;
  flex-grow: 0;
```

Reply:
hmm yes, that fixes the squishing, but then there’s overflow, so I should add truncation too, thanks

![image](https://user-images.githubusercontent.com/43280/141184778-111fd54c-7106-4e91-901f-e4973b2e933f.png)

Reply:
Actually I’m going to just open an issue for that, I don’t know how to handle it right now.